Android/app/src/main/java/com/example/androidgui/manageractivities/SelectEditOrDeleteActivity.java

```diff
@@ -53,16 +69,34 @@ protected void onCreate(Bundle savedInstanceState) {
      * @param v android view.
      */
     @RequiresApi(api = Build.VERSION_CODES.KITKAT)
-    public void manageMenu(View v){
+    public void manageMenu(View v) {
         String action = selectOption[selectEditOrDelete.getValue()];
         Intent extras = getIntent();
         String dishName = extras.getStringExtra("dishSelected");
-        if (Objects.equals(action,ManagerDecision.EDIT.toString())){
-            menuPresenter.deleteDishByName(dishName);
-        }
-        else {
-            menuPresenter.editDishByName(dishName);
+        if (Objects.equals(action, ManagerDecision.EDIT.toString())) {
```

Comment:
long if/else block
extract AlertDialog and pass in title/message/button listener